File: backend/src/main/java/com/aibro/model/Contribution.java
package com.aibro.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;


@Entity
@Table(name = "contributions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
/**
 * JPA entity representing a single contribution within a session, either
 * from a human participant or the AI assistant.
 */
public class Contribution {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "session_id", nullable = false)
    private BrainstormingSession session;

    @Column(nullable = false)
    private String speaker; // participant name or "AI"

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ContributionType type;

    @Column
    private Double confidence; // for voice recognition confidence

    /**
     * Distinguishes between human and AI-generated contributions.
     */
    public enum ContributionType {
        HUMAN, AI
    }
}

File: flutter_app/lib/models/contribution.dart
/// Immutable model representing a single contribution within a session.
class Contribution {
  final int id;
  final String speaker;
  final String content;
  final DateTime timestamp;
  final String type;
  final double? confidence;

  /// Creates a new [Contribution] instance.
  Contribution({
    required this.id,
    required this.speaker,
    required this.content,
    required this.timestamp,
    required this.type,
    this.confidence,
  });

  /// Builds a [Contribution] from a JSON map.
  factory Contribution.fromJson(Map<String, dynamic> json) {
    return Contribution(
      id: json['id'] as int,
      speaker: json['speaker'] as String,
      content: json['content'] as String,
      timestamp: DateTime.parse(json['timestamp'] as String),
      type: json['type'] as String,
      confidence: (json['confidence'] as num?)?.toDouble(),
    );
  }
}
